android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCompleteCarePlanWorker.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021-2023 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.engine.task
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.google.android.fhir.FhirEngine
+import com.google.android.fhir.search.search
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import org.hl7.fhir.r4.model.CarePlan
+import org.hl7.fhir.r4.model.ResourceType
+import org.hl7.fhir.r4.model.Task
+import org.smartregister.fhircore.engine.util.extension.extractId
+
+@HiltWorker
+class FhirCompleteCarePlanWorker
+@AssistedInject
+constructor(
+  @Assisted val context: Context,
+  @Assisted workerParams: WorkerParameters,
+  val fhirEngine: FhirEngine,
+  val fhirCarePlanGenerator: FhirCarePlanGenerator
+) : CoroutineWorker(context, workerParams) {
+  override suspend fun doWork(): Result {
+    fhirEngine
+      .search<CarePlan> {
+        filter(
+          CarePlan.STATUS,
+          { value = of(CarePlan.CarePlanStatus.DRAFT.toCode()) },
+          { value = of(CarePlan.CarePlanStatus.ACTIVE.toCode()) },
+          { value = of(CarePlan.CarePlanStatus.ONHOLD.toCode()) },
+          { value = of(CarePlan.CarePlanStatus.ENTEREDINERROR.toCode()) },
+          { value = of(CarePlan.CarePlanStatus.UNKNOWN.toCode()) }
+        )
+      }
+      .forEach carePlanLoop@{ carePlan ->
+        var shouldCompleteCarePlan = true
+        carePlan
+          .activity
+          .flatMap { it.outcomeReference }
+          .filter { it.reference.startsWith(ResourceType.Task.name) }
+          .mapNotNull { fhirCarePlanGenerator.getTask(it.extractId()) }
+          .forEach { task ->
+            shouldCompleteCarePlan =
+              (task.status in listOf(Task.TaskStatus.CANCELLED, Task.TaskStatus.COMPLETED))
+
+            if (!shouldCompleteCarePlan) return@carePlanLoop
```

Comment:
I don't think the var is necessary because you are always resetting it in the loop
```suggestion
        carePlan
          .activity
          .flatMap { it.outcomeReference }
          .filter { it.reference.startsWith(ResourceType.Task.name) }
          .mapNotNull { fhirCarePlanGenerator.getTask(it.extractId()) }
          .forEach { task ->
            if (task.status !in listOf(Task.TaskStatus.CANCELLED, Task.TaskStatus.COMPLETED)) return@carePlanLoop
```